### PR TITLE
fix throw exception on read stream

### PR DESCRIPTION
**Pull request by casahugo**
fix https://github.com/Nyholm/psr7/issues/153

### CODE REVIEW

File: tests/StreamTest.php [hunk 1]
```diff
@@ -66,6 +66,15 @@ public function testChecksEof()
         $stream->close();
     }
 
+    public function testReadError()
+    {
+        $this->expectException(\RuntimeException::class);
+        $handle = fopen('php://temp', 'a');
```
Comment (mahagr) [inline tests/StreamTest.php:72]: Well, that does work. You cannot read from the write-only stream. :)

Comment (casahugo) [reply tests/StreamTest.php:72]: Yes it's seem work, but why the CI throw an error on BC Check ?

**Comment by Zegnat** [comment]
The newly added test case here does not test the previously wrong behaviour. Note that the following passes for append-only streams:

```php
        $handle = fopen('php://temp', 'a');
        $stream = Stream::create($handle);
        $this->assertFalse($stream->isReadable()); // ← important bit
```

This means the `RuntimeException` you are testing for in the new `testReadError` is actually the existing “Cannot read from non-readable stream” exception and not the newly added “Unable to read from stream”.

The new test case should probably be as follows, checking for the exact error:

```php
    public function testReadError()
    {
        $this->expectException(\RuntimeException::class);
        $this->expectExceptionMessage('Unable to read from stream');
        $handle = fopen('php://temp', 'a');
        $stream = Stream::create($handle);

        $stream->read(4);
    }
```

This will not pass, because the code there is not enough to trigger a return value of `false` from the `fread` :( 

So I am 👍 on the fix, which is the same [as I mentioned](https://github.com/Nyholm/psr7/issues/153#issuecomment-696855655), but the test case remains a head scratcher.

File: tests/StreamTest.php [hunk 2]
```diff
@@ -66,6 +66,15 @@ public function testChecksEof()
         $stream->close();
     }
 
+    public function testReadError()
+    {
+        $this->expectException(\RuntimeException::class);
```
Comment (Zegnat) [inline tests/StreamTest.php:71]: Use the following assertion to make sure you are testing the `RuntimeException` that is being introduced by this PR:
```suggestion
        $this->expectException(\RuntimeException::class);
        $this->expectExceptionMessage('Unable to read from stream');
```
For me this sadly breaks the test 😞 So doing a `->read(4)` from an append-only handle is not enough to trigger the `false` value.

**Review by Zegnat — CHANGES_REQUESTED** [review]
Lets either fix the test or remove it from this PR.

File: tests/StreamTest.php (hunk 2 above)
Comment (casahugo) [reply tests/StreamTest.php:71]: I have no idea so that fread return false :disappointed:, just remove this test.

Comment (Zegnat) [reply tests/StreamTest.php]: Thanks for the update!

**Review by Zegnat — APPROVED** [review]

**Comment by Zegnat** [comment]
All looks good to me 🚀 

I just need to find a minute of extra time to see why the BC checks are failing before hitting merge. Thanks for the quick turnover @casahugo!

**Review by Nyholm — APPROVED** [review]
Thank you for the work and reviews.

I think it all looks good. The BC check is unrelated, see #155

**Pull request merged**